test/experiment/parallelism.cpp

```diff
@@ -0,0 +1,82 @@
+#include <gtest/gtest.h>
+#include <cmath>
+#include <chrono>
+#include <omp.h>
+#include "../../include/supernode.h"
+
+const long seed = 7000000001;
+const unsigned long long int num_nodes = 2000;
+
+class EXPR_Parallelism : public testing::Test {
+protected:
+  static vector<Edge>* graph_edges;
+  static vector<Edge>* odd_graph_edges;
+  static bool* prime;
+  static void SetUpTestSuite() {
+    srand(1000000007);
+    graph_edges = new vector<Edge>();
+    odd_graph_edges = new vector<Edge>();
+    for (unsigned i=2;i<num_nodes;++i) {
+      for (unsigned j = i*2; j < num_nodes; j+=i) {
+        graph_edges->push_back({i,j});
+        if ((j/i)%2) odd_graph_edges->push_back({i,j});
+      }
+    }
+
+    // sieve
+    prime = (bool*) malloc(num_nodes*sizeof(bool));
+    fill(prime,prime+num_nodes,true);
+    for (unsigned i = 2; i < num_nodes; i++) {
+      if (prime[i]) {
+        for (unsigned j = i * i; j < num_nodes; j += i) prime[j] = false;
+      }
+    }
+  }
+  static void TearDownTestSuite() {
+    delete graph_edges;
+    delete odd_graph_edges;
+    free(prime);
+  }
+
+  void SetUp() override {}
+  void TearDown() override {}
+};
+
+vector<Edge>* EXPR_Parallelism::graph_edges;
+vector<Edge>* EXPR_Parallelism::odd_graph_edges;
+bool* EXPR_Parallelism::prime;
```

Comment:
I can't find anywhere this is referenced, do we really need this test fixture?

Reply:
probably not, I'll get rid of it